### ISSUE TITLE
Portal is not ssr compat - accesses `document` during render

### ISSUE THREAD

**Issue by mattcosta7** [primer/react#3721]
### Description

When using `InlineAutocomplete` we inherit usage of `<Portal />` rendering on page load. 

I think the issue occurs because `<Portal>` calls `document.createElement` and `createPortal` with a ref to an element that can't be created. 

https://github.com/primer/react/blob/500e529df47ae5def4bf5cedc55fc099251c25a1/src/Portal/Portal.tsx#L68

we should avoid these apis in render in portal (which creates a div on every render) and rely on other mechanisms to sync it

In addition to this, it renders an _Autocomplette which uses `useCombobox` which calls `useLayoutEffect` resulting in a call to `console.error` on the server - https://github.com/primer/react/blob/500e529df47ae5def4bf5cedc55fc099251c25a1/src/drafts/hooks/useCombobox.ts#L141-L157

### Steps to reproduce

Atempt to SSR an app that uses `InlineAutocomplete`

### Version

latest

**Comment by lesliecdubs** [comment]
@mattcosta7 thanks for filing and for putting up some PRs! Are you still looking into a solve, or would you like support?

**Comment by mattcosta7** [comment]
> @mattcosta7 thanks for filing and for putting up some PRs! Are you still looking into a solve, or would you like support? If the latter, can you also give us some info on whether this is blocking a particular product feature?

I don't think I'll have time , but I don't believe it's blocking anything!

**Comment by lesliecdubs** [comment]
Thanks, copy that. We'll move this to the backlog for now, but happy to reassess priority at any point as needed.

**Comment by keithamus** [comment]
Looks like I haven't done my usual "we should use popover instead" so:

Rather than continuing to invest in portals we should be migrating the remaining cases to use popover instead. There is no good reason to use portals now that popover exists.

**Comment by mattcosta7** [comment]
> Looks like I haven't done my usual "we should use popover instead" so:
> 
> Rather than continuing to invest in portals we should be migrating the remaining cases to use popover instead. There is no good reason to use portals now that popover exists.

I agree there, however since it exists currently and isn't marked deprecated, or should probably not cause issues when run in applications in various environments 
So maybe we should deprecate Portal instead and schedule it for removal, then force feature teams to do something else

Or this component could implement some behavior that uses popover as a migration path?

**Comment by keithamus** [comment]
> So maybe we should deprecate Portal instead and schedule it for removal, then force feature teams to do something else

Yes I think this is a great first step. Announcing deprecation of portal. There are currently still Primer React components that use it but I think the faster we deprecate the faster we migrate.

> Or this component could implement some behavior that uses popover as a migration path?

Ideally all components that use portal today would move to a more specialized component like SelectPanel, Dialog, etc.

**Comment by lesliecdubs** [comment]
FYI, I've catalogued this discussion in an issue to deprecate Portal https://github.com/github/primer/issues/4861 and placed it up for discussion in the next planning cycle.
